go/otgrpc/shared.go

```diff
@@ -18,6 +20,7 @@ type metadataReaderWriter struct {
 }
 
 func (w metadataReaderWriter) Set(key, val string) {
+	key = strings.ToLower(key)
```

Comment:
Thank you! I think you can close #5 which tries to solve the same problem of this PR.